metrics/cloudwatch/cloudwatch.go

```diff
@@ -12,81 +13,131 @@ import (
 	"github.com/go-kit/kit/log"
 	"github.com/go-kit/kit/metrics"
 	"github.com/go-kit/kit/metrics/generic"
+	"github.com/go-kit/kit/metrics/internal/lv"
 )
 
 const (
 	maxConcurrentRequests = 20
 )
 
+type Percentiles []struct {
+	s string
+	f float64
+}
+
 // CloudWatch receives metrics observations and forwards them to CloudWatch.
 // Create a CloudWatch object, use it to create metrics, and pass those metrics as
 // dependencies to the components that will use them.
 //
 // To regularly report metrics to CloudWatch, use the WriteLoop helper method.
 type CloudWatch struct {
-	mtx                   sync.RWMutex
-	sem                   chan struct{}
-	namespace             string
-	numConcurrentRequests int
-	svc                   cloudwatchiface.CloudWatchAPI
-	counters              map[string]*counter
-	gauges                map[string]*gauge
-	histograms            map[string]*histogram
+	mtx        sync.RWMutex
+	sem        chan struct{}
+	namespace  string
+	svc        cloudwatchiface.CloudWatchAPI
+	counters   *lv.Space
+	gauges     *lv.Space
+	histograms *lv.Space
+	*cwoptions
+}
+
+type cwoptions struct {
+	percentiles           Percentiles
 	logger                log.Logger
+	numConcurrentRequests int
+}
```

Comment:
Why is this a separate struct embedded into the type? I'd prefer to see these things flattened into the parent.